src/main/java/com/zilen/weather/service/WeatherService.java

```diff
@@ -25,13 +27,21 @@ public WeatherService(RestTemplate restTemplate, WeatherRepository weatherReposi
         this.weatherConverter = weatherConverter;
     }
 
+    public Weather findByCityNameFromApi(String cityName) {
+        ResponseEntity<Weather> response = restTemplate.getForEntity(url + cityName + "&units=metric&appid=" + appId, Weather.class);
+        weatherRepository.save(weatherConverter.transformToEntity(response.getBody()));
+        return weatherConverter.transformToWeather(weatherRepository.findByName(cityName));
```

Comment:
Right here you make a two calls to db, `save` method of repository return entity that you need. Please remove `findByName` call.